model.py

```diff
@@ -6616,6 +6620,98 @@ def update_availability(
 
         return changes_made
 
+    def update_availability_from_delta(self, event_type, event_date, delta, analytics=None):
+        """Call update_availability based on a single change seen in the
+        distributor data, rather than a complete snapshot of
+        distributor information as of a certain time.
+
+        This information is unlikely to be completely accurate, but it
+        should suffice until more accurate information can be
+        obtained.
+
+        No CirculationEvent is created until `update_availability` is
+        called.
+
+        Events must be processed in chronological order. Any event
+        that happened than `LicensePool.last_checked` is ignored, and
+        calling this method will update `LicensePool.last_checked` to
+        the time of the event.
+
+        :param event_type: A CirculationEvent constant representing the
+        type of change that was seen.
+
+        :param event_date: A datetime corresponding to when the 
+        change was seen.
+
+        :param delta: The magnitude of the change that was seen.
+
+        """
+        if event_date != CirculationEvent.NO_DATE and self.last_checked and event_date < self.last_checked:
+            # This is an old event and its effect on availability has
+            # already been taken into account.
+            return
+
+        if self.last_checked and event_date == CirculationEvent.NO_DATE:
+            # We have a history for this LicensePool and we don't know
+            # where this event fits into that history. Ignore the
+            # event.
+            return
+
+        (new_licenses_owned, new_licenses_available, 
+         new_licenses_reserved, 
+         new_patrons_in_hold_queue) = self._calculate_change_from_one_event(
+             event_type, delta
+         )
```

Comment:
`update_availability` uses these four values to guess the event type, basically reversing the work you just did. I'm not sure how to improve it though.